lighthouse-core/audits/third-party-summary.js

```diff
@@ -57,41 +85,103 @@ class ThirdPartySummary extends Audit {
    * @param {Array<LH.Artifacts.NetworkRequest>} networkRecords
    * @param {Array<LH.Artifacts.TaskNode>} mainThreadTasks
    * @param {number} cpuMultiplier
-   * @return {Map<ThirdPartyEntity, {mainThreadTime: number, transferSize: number, blockingTime: number}>}
+   * @return {{byEntity: Map<ThirdPartyEntity, Summary>, byURL: Map<string, Summary>, urls: Map<ThirdPartyEntity, string[]>}}
    */
-  static getSummaryByEntity(networkRecords, mainThreadTasks, cpuMultiplier) {
-    /** @type {Map<ThirdPartyEntity, {mainThreadTime: number, transferSize: number, blockingTime: number}>} */
-    const entities = new Map();
-    const defaultEntityStat = {mainThreadTime: 0, blockingTime: 0, transferSize: 0};
+  static getSummaries(networkRecords, mainThreadTasks, cpuMultiplier) {
+    /** @type {Map<string, Summary>} */ const byURL = new Map();
+    /** @type {Map<ThirdPartyEntity, Summary>} */ const byEntity = new Map();
+    const defaultStat = {mainThreadTime: 0, blockingTime: 0, transferSize: 0};
 
     for (const request of networkRecords) {
-      const entity = thirdPartyWeb.getEntity(request.url);
-      if (!entity) continue;
-
-      const entityStats = entities.get(entity) || {...defaultEntityStat};
-      entityStats.transferSize += request.transferSize;
-      entities.set(entity, entityStats);
+      const urlStat = byURL.get(request.url) || {...defaultStat};
+      urlStat.transferSize += request.transferSize;
+      byURL.set(request.url, urlStat);
     }
 
     const jsURLs = BootupTime.getJavaScriptURLs(networkRecords);
 
     for (const task of mainThreadTasks) {
-      const attributeableURL = BootupTime.getAttributableURLForTask(task, jsURLs);
-      const entity = thirdPartyWeb.getEntity(attributeableURL);
-      if (!entity) continue;
+      const attributableURL = BootupTime.getAttributableURLForTask(task, jsURLs);
 
-      const entityStats = entities.get(entity) || {...defaultEntityStat};
+      const urlStat = byURL.get(attributableURL) || {...defaultStat};
```

Comment:
the fn that makes this is "getSummaries" but the value here (and within that function) is "urlStat". Let's rename the values or the function to match. I suggest keeping the fn name and using `urlSummary` instead of `stat`.